.github/workflows/build.yml

```
@@ -247,9 +247,18 @@ jobs:
 
     - name: Upload self-contained executable
       uses: actions/upload-artifact@b7c566a772e6b6bfb58ed0dc250532a479d7789f # v6.0.0
-      if: (github.ref_name == 'main' || github.ref_name == 'release') && matrix.PUBLISH_RELEASE && !env.ACT
+      if: |
+        (matrix.PUBLISH_RELEASE && !env.ACT) &&
+        ((github.ref_name == 'main' || github.ref_name == 'release') || github.event_name == 'workflow_dispatch')
       with:
-        name: artifacts-${{ matrix.os }}
+        # Branch-suffix is only appended for workflow_dispatch runs on non-main/non-release branches.
+        # Slashes in branch names are replaced with dashes. This ensures artifacts from main/release
+        # remain unsuffixed and avoid collisions. See the outer if: condition that gates uploads.
```

actionlint (1.7.10)

could not parse as YAML: did not find expected key

(syntax-check)